installers/rpm-code/codes/mantid.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+mantid_main() {
+    codes_yum_dependencies \
+        eigen3-devel \
+        OCE-devel \
+        jemalloc-devel \
+        jsoncpp-devel \
+        muParser-devel \
+        poco-devel
+    codes_dependencies common boost ipykernel
+    install_pip_install pre-commit pylint sphinx_bootstrap_theme
+    mantid_nexus_install
+    # Mantid installs a great deal and requires its own specific environment variables to run right
+    # (ex. see the mantidpython script it creates). So, install it in its own subdirectory so it
+    # doesn't interfere with other libs.
+    local p="${codes_dir[lib]}/mantid"
+    mantid_install "$p"
+    mantid_ipykernel $(mantid_install_rsmantid "$p")
+}
+
+mantid_install() {
+    local install_prefix="$1"
+    codes_download mantidproject/mantid
+    mantid_patch_eigen_cmake
+    codes_cmake \
+        -DBOOST_ROOT="${codes_dir[prefix]}" \
+        -DCMAKE_INSTALL_PREFIX="$install_prefix" \
+        -DENABLE_WORKBENCH=OFF \
+        -DMANTID_FRAMEWORK_LIB=BUILD \
+        -DMANTID_QT_LIB=OFF \
+        -DNEXUS_CPP_LIBRARIES="${codes_dir[lib]}/libNeXusCPP.so" \
+        -DNEXUS_C_LIBRARIES="${codes_dir[lib]}/libNeXus.so" \
+        -DNEXUS_INCLUDE_DIR="${codes_dir[include]}/nexus/" \
+        -G'Unix Makefiles'
+    codes_cmake_build
+    codes_make_install
+}
+
+mantid_install_rsmantid() {
+    local s=rsmantid
+    codes_download_module_file "$s.sh"
+    local l="$1/bin/mantidpython"
+    MANTID_PYTHON_SCRIPT_INSTALL_LOCATION="$l" \
+    perl -p -e 's/\$\{(\w+)\}/$ENV{$1} || die("$1: not found")/eg' "$s.sh" \
+        | install -m 555 /dev/stdin "${codes_dir[bin]}"/"$s"
+    echo "$l"
+}
+
+mantid_ipykernel() {
+    # Needs to be in same dir as $1 because script uses relative paths
+    local p="$1-$RANDOM"
+    patch --quiet --output="$p" "$1" <<'EOF'
+@@ -54,5 +54,5 @@
+ fi
+
+-LD_PRELOAD=${LOCAL_PRELOAD} \
++echo LD_PRELOAD=${LOCAL_PRELOAD} \
+     PYTHONPATH=${LOCAL_PYTHONPATH} \
+     QT_API=${LOCAL_QT_API} \
+EOF
+    local -a e=( $(bash "$p"))
+    e=( "${e[@]::${#e[@]}-5}" )
```

Comment:
You don't need to rewrite e. Just put it in place. Less code. 

However, this is where a little copying than a lot of code. There's a lot of complexity in what's going on. Do we know the basic relative dirs for the paths and can set PYTHONPATH LD_PRELOAD (and LD_LIBRARY_PATH?) manually without all these evals?

Reply:
Can you give me the syntax for that? `for v in ( ${e[@]::${#e[@]}-5} ); do` doesn't work.

What do you mean by `basic relative dirs`? I can find out what PYTHONPATH etc end up being set to and set them myself but I preferred this way because:
1. If they change the script to add new values maybe we'll pick them up
2. If they change the script maybe the patch will break and we'll be tipped off that something has changed
3. It brings the readers attention to the mantidpython script and that it holds information about what environment is needed to get mantid running. So, it is a breadcrumb to help with debugging in the future. 

Obviously none of that is foolproof but I think it is better than the flipside which is we'll recompile mantid in the future, it will build fine but someone will tell us it isn't working right, we'll have to trace it back down to some environment issue and I'll have to ask myself how I determined this environment in the first place.

Reply:
Can you attach a copy of mantidpython after it has been generated to here? I'd like to see if what's going on is what I think's going on. :)

Reply:
[Here](https://github.com/radiasoft/download/files/8993130/mantidpython.txt) it is. FWIW, [this is the template](https://github.com/radiasoft/download/files/8993140/mantidpython.in.txt) that file is generated from.

Reply:
`for v in ( ${e[@]::${#e[@]}-5} ); do` is too much try `for v in "${e[@]::${#e[@]}-5}"`.  Parens for arrays are only meaningful in some contexts.